syntax="proto2";

package nbtp;

message ClientAuth {
  required string passphrase = 1;
}

message ServerReportAuth {
  required bool AuthPassed = 1;
}

message ClientSetTarget {
  required string target_url = 1;
  repeated string headers = 2;
}

message ServerReportTarget {
  optional string connection_error_message = 1;
  optional uint64 file_size = 2;
  required int32 mission_id = 3;
}

message ClientGetHistory {// server will report a ServerReportAllChunks packet.
  required int32 mission_id = 1;
}

message ServerChunkStatus { // describe a unit_status object in reporter
  required int32 uploading_status = 1;// 0-wait 1-uploading 2-complete
  required int32 cid = 2;
  optional uint64 chunk_length = 3; // if no length, the mission haven't start yet.
  required uint64 already_uploaded_size = 4;
  required int32 failed_times = 5;
  optional string last_fail_reason = 6; // if fail times is zero, this is not set.
  optional string result_key = 7; // the final result link.
}

message ServerChunkList {
  repeated ServerChunkStatus status_list = 1;
}

message ServerReport {
  oneof ServerReportMessage {
    ServerChunkStatus status = 1;
    ServerChunkList status_list = 2;
  }
}






